benefits/oauth/views.py

```diff
@@ -0,0 +1,44 @@
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.utils.decorators import decorator_from_middleware
+
+from authlib.integrations.django_client import OAuth
+
+from benefits.core import middleware, session
+
+
+_oauth = OAuth()
+
+
+@decorator_from_middleware(middleware.AgencySessionRequired)
+@decorator_from_middleware(middleware.VerifierSessionRequired)
+def login(request):
+    verifier = session.verifier(request)
+    if not verifier.requires_authentication:
+        return redirect(session.origin(request))
+
+    oauth_client = _oauth.create_client(verifier.auth_provider.name)
+
+    if not oauth_client:
+        return redirect(session.origin(request))
```

Comment:
I don't think there's anything wrong with doing this redirect, but would it be more appropriate to raise an exception instead? It seems like we'd get into this branch if we forgot to add the auth provider's configuration to `.env` - i.e. this is a state we don't expect to be in.

Reply:
Hmm... yeah this definitely indicates a bad configuration, mismatch between the fixtures and settings.py or something. Exception is probably a better idea! 👍

Reply:
Should we have a fallback one? Seems a bit strange that someone getting to the `login` URL (one way or another) might not actually be logged in.

Reply:
@afeld Do you mean a fallback that results in the user effectively being seen as logged in? I'm not sure I see the need for that since, other than the case that's handled by `if not verifier.requires_authentication:`, we would always want the user to go through a real authentication process. (Maybe I misunderstood your suggestion though?)

Reply:
I was a bit confused by @afeld wording too - I think the question is about having a fallback OAuth client to use, in the case that we can't find the one named by the `auth_provider`.

I'm inclined to throw the Exception rather than try to figure out a fallback client. It really shouldn't happen / be needed, and we should like to know if the configuration is bad.

Agreed that the redirect back out (possibly indicating a successful login) is probably not the best approach.

Reply:
So addressing both of your comments in this thread and @afeld [other related comments](https://github.com/cal-itp/benefits/pull/414#discussion_r847693486):

I moved the `client_name` parameter in `_oauth.create_client()` here to be a setting rather than a property of the verifier. This simplifies the views, to where they don't really have to care about the `session`. We basically have a "on" / "off" for OAuth for the app, rather than per-`verifier`.

This also means means an Exception is thrown if these view functions are called and an OAuth client wasn't configured.